x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -285,8 +286,25 @@ public void validate(ByteSizeValue value, Map<Setting<?>, Object> settings, bool
 
     private final LongAdder evictCount = new LongAdder();
 
+    private final BlobCacheMetrics blobCacheMetrics;
+
+    public SharedBlobCacheService(
+        NodeEnvironment environment,
+        Settings settings,
+        ThreadPool threadPool,
+        String ioExecutor,
+        BlobCacheMetrics blobCacheMetrics
```

Comment:
@stu-elastic discussed two ways of registering metrics upfront (which is the preferred pattern)
1. This way, by passing in a container of references to metrics
2. By declaring the `name` as a final string somewhere, and registering them where this guaranteed to be done once. Then, passing in the `MetricsRegistry` here and using `get*` with a reference to the name to get them here.

I decided against 2 in favor of 1 because 1 moves the guarantee of registration into the "type" level from the `SharedBlobCacheService` perspective. This enforces correctness.  With 2, it's possible to construct a `SharedBlobCacheService` with non-registered metrics, resulting in Null Pointer Exceptions.  This approach does not enforce correctness at the type level.